Source/WebKit/Shared/skia/CoreIPCSkFontStyleSlant.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2024 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if USE(SKIA)
+
+#include <skia/core/SkFontStyle.h>
+
+namespace WebKit {
+
+class CoreIPCSkFontStyleSlant {
+public:
+    CoreIPCSkFontStyleSlant(SkFontStyle::Slant slant)
+        : m_slant(slant)
+    {
+    }
+
+    static SkFontStyle::Slant create(uint8_t slant)
+    {
+        return static_cast<SkFontStyle::Slant>(slant);
```

Comment:
This doesn't prevent out-of-bounds values.  I think it would be better to make an enum class that mirrors Slant's 3 values and use that instead.

Reply:
SkFontStyle constructor clamps the input value with `SkTPin` which is using min and max.
https://github.com/WebKit/WebKit/blob/ac3edb570f403e4e1e47bee55f610a99f9d46c38/Source/ThirdParty/skia/include/core/SkFontStyle.h#L53C10-L53C16
https://github.com/WebKit/WebKit/blob/6194732613da51a04d72cb4b25b670c69ab6e663/Source/ThirdParty/skia/include/private/base/SkTPin.h#L20

Reply:
Yikes